From 86937dc071f3634ba1f9a655684f1320e6514aa1 Mon Sep 17 00:00:00 2001
From: dev@builder
Date: Thu, 16 Jun 2016 16:31:34 +0200
Subject: [PATCH] fix: cpu-probe: disable RIXIEX check on BCM7584

on BCM7584 (and maybe others?) setting the PG_IEC bit causes the second
CPU to stall during init
---
 arch/mips/kernel/cpu-probe.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/mips/kernel/cpu-probe.c b/arch/mips/kernel/cpu-probe.c
index 48dfb9d..f1a64e0 100644
--- a/arch/mips/kernel/cpu-probe.c
+++ b/arch/mips/kernel/cpu-probe.c
@@ -548,6 +548,7 @@ static void decode_configs(struct cpuinfo_mips *c)
 
 	mips_probe_watch_registers(c);
 
+#ifndef CONFIG_BCM7584
 	if (cpu_has_rixi) {
 		/* Enable the RIXI exceptions */
 		set_c0_pagegrain(PG_IEC);
@@ -556,6 +557,7 @@ static void decode_configs(struct cpuinfo_mips *c)
 		if (read_c0_pagegrain() & PG_IEC)
 			c->options |= MIPS_CPU_RIXIEX;
 	}
+#endif
 
 #ifndef CONFIG_MIPS_CPS
 	if (cpu_has_mips_r2_r6) {
-- 
1.9.1

